dashboard/app/models/script.rb

```diff
@@ -228,6 +249,17 @@ def self.level_cache
     end
   end
 
+  def self.update_level_in_cache(level)
+    return unless self.should_cache?
+    level_cache[level.id] = level if level_cache.key? level.id
+    level_cache[level.name] = level if level_cache.key? level.name
+  end
+
+  def self.delete_level_from_cache(level)
+    return unless self.should_cache?
+    @@level_cache.except!(level.id, level.name)
+  end
```

Comment:
Moved these methods, along with the other level cache and script level cache methods into their respective models. Also added an after_destroy handler for scripts.